### PR TITLE
Subclass abcmeta

### PR DESCRIPTION
**Pull request by jcugat**
Implemented what we discussed in #120

Also changed pytest execution in the Makefile (was failing in my machine).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -19,7 +19,7 @@ lint:
 
 .PHONY: test
 test:
-	pytest --cov=pydantic
```
Comment (samuelcolvin) [inline Makefile]: remove these changes, the problem should be fixed by Frozenball/pytest-sugar#133 which will hopefully be merged soon.

If the problems persist I'll remove `pytest-sugar` completely.

**Comment by jcugat** [comment]
What kind of documentation are you thinking about? Since it's in the stdlib the documentation is already there: https://docs.python.org/3/library/abc.html

Maybe just a mention in the changelog?

**Comment by samuelcolvin** [comment]
Just a show section in the main docs saying something like "pydantic works with python's standard abc" then a mini example in python.

**Comment by jcugat** [comment]
Tests are failing since I readded `pytest-sugar` to the Makefile.

**Comment by samuelcolvin** [comment]
Looks great. Thank you very much. I'll wait a couple of days in the hope that pytest-sugar gets fixed, if it doesn't I'll remove it.

Once that's done this looks ready to merge.

**Comment by samuelcolvin** [comment]
great, thank you very much.

**Pull request merged**